### ISSUE TITLE
Ref to markdown anchor broken in 1.9.7

### ISSUE THREAD

**Issue by lelegard** [doxygen/doxygen#10230]
Starting with doxygen version 1.9.7, referencing an anchor in a markdown file from a dox file no longer works.

Output:
~~~
mainpage.dox:3: warning: unable to resolve reference to 'building' for \ref command
~~~

File `building.md`:
~~~
# Building project {#building}
~~~

File `mainpage.dox`:
~~~
/**
@mainpage Main page
@ref building "Building project"
*/
~~~

The set of files to reproduce the issue, including `Doxyfile`, is attached in [doxybug.zip](https://github.com/doxygen/doxygen/files/12332773/doxybug.zip).

The problem appears in version 1.9.7. After downloading and build previous versions:
~~~
$ ../doxygen-1.9.4/build/bin/doxygen
$ ../doxygen-1.9.5/build/bin/doxygen
$ ../doxygen-1.9.6/build/bin/doxygen
$ ../doxygen-1.9.7/build/bin/doxygen
mainpage.dox:3: warning: unable to resolve reference to 'building' for \ref command
$
~~~

**Comment by albert-github** [comment]
With the doxygen version 1.9.7 I get the warning as indicated.
With the current master version of doxygen  (1.9.8 (a996a63a5a7527b29117737db60019ade81d7fa7)) the warning is gone and the link is back.
(I also can remember that something has been fixed in this respect).

I think the problem is fixed but not released.

**Comment by lelegard** [comment]
Thanks @albert-github

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
